pkg/common/yaml_decoder.go

```
@@ -14,15 +15,22 @@ import (
 
 type DecodeCallback = func(runtime.Object) error
 
+// DecodeFile decodes the provided file data (encoded YAML documents) into Kubernetes objects using the specified scheme,
+// and invokes the callback function for each decoded object. Returns an error if any decoding error occurs.
 func DecodeFile(ctx context.Context, fileData []byte, scheme *runtime.Scheme, cb DecodeCallback) error {
-	logger, _ := logr.FromContext(ctx)
+	logger, logErr := logr.FromContext(ctx)
 	codec := serializer.NewCodecFactory(scheme)
 	decoder := codec.UniversalDeserializer()
 
+	if logErr != nil {
+		return fmt.Errorf("logger is nil")
```

Better to return the actual error instead of creating one